tests/cpp_api/unit_tests/unit_tests.cpp

```diff
@@ -707,6 +713,51 @@ int test(int test_case, int reader_type, const char *path, const char *outName,
                     }
                 }
             } break;
+            case 4: {  // numpy reader pipeline
+                RocalTensorList output_tensor_list;
+                output_tensor_list = rocalGetOutputTensors(handle);
+                for (int idx = 0; idx < output_tensor_list->size(); idx++) {
+                    unsigned char *out_buffer;
+                    if (output_tensor_list->at(idx)->data_type() == RocalTensorOutputType::ROCAL_FP32) {
+                        float *out_f_buffer;
+                        std::cout << "Creating float buffer of ";
+                        for (auto x : output_tensor_list->at(idx)->shape())
+                            std::cout << x << " x ";
+                        std::cout << "shape\n";
+                        if (output_tensor_list->at(idx)->backend() == RocalTensorBackend::ROCAL_GPU) {
+                            out_f_buffer = (float *)malloc(output_tensor_list->at(idx)->data_size());
+                            output_tensor_list->at(idx)->copy_data(out_f_buffer);
+                        } else if (output_tensor_list->at(idx)->backend() == RocalTensorBackend::ROCAL_CPU)
+                            out_f_buffer = (float *)output_tensor_list->at(idx)->buffer();
+
+                        out_buffer = (unsigned char *)malloc(output_tensor_list->at(idx)->data_size() / 4);
+                    } else if (output_tensor_list->at(idx)->data_type() == RocalTensorOutputType::ROCAL_FP16) {
+                        half *out_f16_buffer;
+                        std::cout << "Creating float16 buffer of ";
+                        for (auto x : output_tensor_list->at(idx)->shape())
+                            std::cout << x << " x ";
+                        std::cout << "shape\n";
+                        if (output_tensor_list->at(idx)->backend() == RocalTensorBackend::ROCAL_GPU) {
+                            out_f16_buffer = (half *)malloc(output_tensor_list->at(idx)->data_size());
+                            output_tensor_list->at(idx)->copy_data(out_f16_buffer);
+                        } else if (output_tensor_list->at(idx)->backend() == RocalTensorBackend::ROCAL_CPU)
+                            out_f16_buffer = (half *)output_tensor_list->at(idx)->buffer();
+
+                        out_buffer = (unsigned char *)malloc(output_tensor_list->at(idx)->data_size() / 2);
+                    } else {
```

Comment:
These if else conditions for different datatype may not be required, 
Please combine these into 1, only few vars are changing, The ptr can be void * itself I think
Please check this

Reply:
Please check latest version and update if changes are still needed